vllm/entrypoints/openai/serving_engine.py

```diff
@@ -153,6 +167,23 @@ def _maybe_get_adapters(
         for prompt_adapter in self.models.prompt_adapter_requests:
             if request.model == prompt_adapter.prompt_adapter_name:
                 return None, prompt_adapter
+        if request.model and envs.VLLM_ADAPTER_DIR is not None:
+            adapter_path = os.path.join(envs.VLLM_ADAPTER_DIR, request.model)
+            adapter_config_path = os.path.join(adapter_path,
+                                               "adapter_config.json")
+            if os.path.exists(adapter_config_path):
+                with open(adapter_config_path) as file:
+                    adapter_config = json.load(file)
+                if adapter_config["peft_type"] == "LORA":
+                    lora_request = LoadLoRAAdapterRequest(
+                        lora_name=request.model,
+                        lora_path=adapter_path,
+                    )
+                    response = await self.models.load_lora_adapter(lora_request
+                                                                   )
+                    if isinstance(response, ErrorResponse):
+                        raise ValueError(response.message)
+                return None, None
```

Comment:
On success, `self.models.load_lora_adapter` should cache a new `LoRARequest` for this adapter in `self.models.lora_requests`. Shouldn't we return that here, instead of `None, None`?

I guess if this logic block was moved above line 164 here and the return wsa removed, then the existing block that checks for lora adapters would find and return it:
```
        for lora in self.models.lora_requests:
            if request.model == lora.lora_name:
                return lora, None
```

Reply:
so, you're not wrong, but then why is the test I wrote passing...?